### PR TITLE
Update to run with ODB 2.1.6 and some smaller changes

### PR DESCRIPTION
**Pull request by smolinari**
This has a number of changes, so the ODB studio also works properly.

Scott

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,6 @@
-composer.lock
```
Comment (PommeVerte) [inline .gitignore]: this shouldn't be removed. instead it should be updated with `composer update <vendor>/<package>`

**Comment by dmill-bz** [comment]
This looks good. Lets see what @chrismichaels84 has to say

**Comment by smolinari** [comment]
Ok. I've also cleaned it up. I hope it is mergeable. :smile: 

Scott

**Comment by electricjones** [comment]
@smolinari This looks good to me, but I can't test it right now. My internet isn't so great atm. I'll have a pull and run vagrant tomorrow just to be sure.

@PommeVerte, it would be nice if vagrant and Travis shared some bootstrapping. That  way we only have to change versions in one spot. I tried (for a few minutes) but had troubles with working directories. Any thoughts? Is this even worth the effort?

**Comment by dmill-bz** [comment]
Yeah I think it's worth the effort. I didn't have a proper look at the vagrant build but I think a few things can be done.

**Comment by electricjones** [comment]
This looks great. Merging.

**Pull request merged**